### PR TITLE
Improvements to generator of conversion methods.

### PR DESCRIPTION
**Pull request by wojtek-t**
- generate conversion functions in alphabetical order
- better handling of pointer arguments
- some small refactoring

Part of #6800 

cc @smarterclayton @lavalamp

### CODE REVIEW

**Comment by wojtek-t** [comment]
@lavalamp you suggested adding a test to checking that functions registered via AddGeneratedConversionFunctions() weren't manually modified. I completely agree this is a good idea but I started thinking what's the best way to do it. This will be even harder as soon as we change those functions to be named and calling them one from another since they will need to be defined outside "AddGeneratedConversionFunctions()" method. Any suggestions on wht is the best way to do it? Do we want to explicitly compare the file as text with what would be generated? Thanks!

**Comment by lavalamp** [comment]
@wojtek-t Yeah, I think auto-generated stuff has to be clearly demarcated in files such that that test can just be a direct comparison. This is also important for humans editing things, so they know to edit the source and not the autogenerated stuff.

File: pkg/conversion/generator.go [hunk 1]
```diff
@@ -340,10 +364,62 @@ func (g *generator) writeConversionForSlice(w io.Writer, inField, outField refle
...
+			return nil
+		}
+	}
+
+	assignFormat := "if err := s.Convert(&in.%s, &out.%s, 0); err != nil {\n"
+	assignStmt := fmt.Sprintf(assignFormat, inField.Name, outField.Name)
+	if err := writeLine(w, indent, assignStmt); err != nil {
 		return err
 	}
+	if err := writeLine(w, indent+1, "return err\n"); err != nil {
```
Comment (lavalamp) [inline pkg/conversion/generator.go]: Check out http://golang.org/pkg/go/format/#Source for auto-formatting of this code. Then you could ignore tabbing.

File: pkg/conversion/generator.go [hunk 2]
```diff
@@ -340,10 +364,62 @@ func (g *generator) writeConversionForSlice(w io.Writer, inField, outField refle
...
+		// This should handle pointers to all simple types.
+		assignable := inField.Type.Elem().AssignableTo(outField.Type.Elem())
+		convertible := inField.Type.Elem().ConvertibleTo(outField.Type.Elem())
+		if assignable || convertible {
+			ifFormat := "if in.%s != nil {\n"
+			ifStmt := fmt.Sprintf(ifFormat, inField.Name)
+			if err := writeLine(w, indent, ifStmt); err != nil {
+				return err
+			}
+			newFormat := "out.%s = new(%s)\n"
```
Comment (lavalamp) [inline pkg/conversion/generator.go]: Maybe ```if out.%s == nil { out.%s = new(%s) } \n``` ?

**Comment by lavalamp** [comment]
LGTM -- comments are optional and can be addressed in a followup

**Pull request merged**